src/extension/content/injected.ts

```diff
@@ -206,3 +293,50 @@ const Netflix = {
     isPlayerReady: Netflix.getPlayer()?.isReady?.() || false,
   }),
 };
+
+const VK = {
+  getPlayer: () => (window as any)?.getAudioPlayer() ?? null,
+  getInfo(): VKInfo {
+    let player = VK.getPlayer();
+    return {
+      state: !player?.getCurrentPlaylist() ? StateMode.STOPPED : player.isPlaying() ? StateMode.PLAYING : StateMode.PAUSED,
+      title: VK.unescape(player?.stats?.currentAudio?.title ?? ""),
+      artist: VK.unescape(player?.stats?.currentAudio?.performer ?? ""),
+      cover: player?.stats?.currentAudio?.coverUrl_p ?? "",
+      duration: player?.stats?.currentAudio?.duration ?? 0,
+      position: Math.floor((player?.getCurrentProgress() ?? 0) * (player?.stats?.currentAudio?.duration ?? 0)),
+      volume: Math.floor((player?.getVolume() ?? 1) ** (1 / 3) * 100),
+      repeatMode: player?.isRepeatAll() ? RepeatMode.ALL : player?.isRepeatCurrentAudio() ? RepeatMode.ONE : RepeatMode.NONE,
+      shuffleActive: player?.getPlaylistQueue()?.shuffled ?? false,
+
+      isPlayerReady: player !== null,
+    };
+  },
+  setState(state: StateMode) {
+    switch (state) {
+      case StateMode.PAUSED:
+        return VK.getPlayer()?.pause();
+      case StateMode.STOPPED:
+        return VK.getPlayer()?.stop();
+      case StateMode.PLAYING:
+        return VK.getPlayer()?.play();
+    }
+  },
+  skipPrevious: () => VK.getPlayer()?.playPrev(),
+  skipNext: () => VK.getPlayer()?.playNext(),
+  setPosition: (pos: number) => VK.getPlayer()?.seekToTime(pos),
+  setVolume: (vol: number) => VK.getPlayer()?.setVolume((vol / 100) ** 3),
+  toggleRepeatMode() {
+    let player = VK.getPlayer();
+    if (player?.isRepeatAll()) return player.toggleRepeatCurrentAudio();
+    if (player?.isRepeatCurrentAudio()) return player.toggleRepeatCurrentAudio();
+    return player?.toggleRepeatAll();
+  },
```

Comment:
I assume this works correctly but just in case, it should toggle through all modes, one mode every time it is called. Eg. none -> one -> all -> none ...

Reply:
No that's fine.